drivers/pinctrl/pinctrl_mcux_rt.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2021 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/pinctrl.h>
+#include <soc.h>
+#include <fsl_iomuxc.h>
+#include <fsl_gpio.h>
+
+int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt,
+			   uintptr_t reg)
+{
+	/* configure all pins */
+	for (uint8_t i = 0U; i < pin_cnt; i++) {
+		uint32_t mux_register = pins[i].pinmux.mux_register;
+		uint32_t mux_mode = pins[i].pinmux.mux_mode;
+		uint32_t input_register = pins[i].pinmux.input_register;
+		uint32_t input_daisy = pins[i].pinmux.input_daisy;
+		uint32_t pin_ctrl_flags = pins[i].pin_ctrl_flags;
+		volatile uint32_t *config_register = (uint32_t *)pins[i].pinmux.config_register;
+		uint32_t config_val = *config_register;
+
+
+
+		IOMUXC_SetPinMux(mux_register, mux_mode, input_register,
+			input_daisy, (uint32_t)config_register,
+			MCUX_RT_INPUT_ENABLE(pin_ctrl_flags));
+
+		if (MCUX_RT_INPUT_SCHMITT_ENABLE(pin_ctrl_flags)) {
```

Comment:
I think below judgement is not necessary, as you have shift the pin_ctrl_flags in pinctrl_soc_mcux_rt_common.h, where the dts setting is translated to.

Reply:
Those shifts are chosen arbitrarily, and don't correspond to the actual bitfield location in the control register. If I was careful when selecting the shift offset of these one bit fields, then I could use the value in `pin_ctrl_flags` directly. Even with this, some of the `if` statements are required, because for some fields there is a special "none" value that tells the driver not to configure the property for that pin and leave it at the SOC default.

That being said, for these one bit fields the `if` statement is not necessary, and can be replaced with something like the following:
```
config_val |= IOMUXC_SW_PAD_CTL_PAD_HYS(MCUX_RT_INPUT_SCHMITT_ENABLE(pin_ctrl_flags));
```
Thanks for pointing that out, I'll update the PR accordingly